src/vos/vos_dtx_iter.c

```diff
@@ -120,19 +120,42 @@ static int
 dtx_iter_next(struct vos_iterator *iter)
 {
 	struct vos_dtx_iter	*oiter = iter2oiter(iter);
+	struct vos_dtx_act_ent	*dae;
+	d_iov_t			 rec_iov;
+	int			 rc = 0;
 
 	D_ASSERT(iter->it_type == VOS_ITER_DTX);
 
-	return dbtree_iter_next(oiter->oit_hdl);
+	while (1) {
+		rc = dbtree_iter_next(oiter->oit_hdl);
+		if (rc != 0)
+			break;
+
+		d_iov_set(&rec_iov, NULL, 0);
+		rc = dbtree_iter_fetch(oiter->oit_hdl, NULL, &rec_iov, NULL);
+		if (rc != 0)
+			break;
+
+		D_ASSERT(rec_iov.iov_len == sizeof(struct vos_dtx_act_ent));
+		dae = (struct vos_dtx_act_ent *)rec_iov.iov_buf;
+
+		/* Only need to return the DTX that was handled before the
+		 * latest DTX resync.
+		 */
+		if (DAE_SRV_GEN(dae) < oiter->oit_cont->vc_dtx_resync_gen)
```

Comment:
would it be better to check this by caller?

Reply:
Then we have to maintain another resync_gen outside of VOS.